go-controller/pkg/cluster/master.go

```diff
@@ -279,13 +279,16 @@ func (cluster *OvnClusterController) addNode(node *kapi.Node) (err error) {
 		return err
 	}
 
-	// Set the HostSubnet annotation on the node object to signal
-	// to nodes that their logical infrastructure is set up and they can
-	// proceed with their initialization
-	err = cluster.Kube.SetAnnotationOnNode(node, OvnHostSubnet, hostsubnet.String())
-	if err != nil {
-		logrus.Errorf("Failed to set node %s host subnet annotation: %v", node.Name, hostsubnet.String())
-		return err
+	if subnetAllocator != nil {
```

Comment:
If `subnetAllocator` is `nil` then shouldn't that imply that `cluster.ensureNodeLogicalNetwork()` has also already been called and can be skipped?

The use of `subnetAllocator` here is pretty confusing. Adding an explanatory comment to `cluster.ensureNodeHostSubnet()` would help a little.

Reply:
Fixed.